seq2seq/models/attention.py

```diff
@@ -73,6 +74,24 @@ def forward(self, decoder_states, encoder_states):
 
         attn = F.softmax(attn.view(-1, input_size), dim=1).view(batch_size, -1, input_size)
 
+        # In the case of hard attentive guidance with variable length examples in a single batch,
+        # The attention will be on the last encoder state. However, the mask will set this to -inf, which will
+        # make all attention scores -inf. Taking the softmax of this results in NaNs. With the following, we set
+        # all NaNs to zero.
+        # Example:
+        # 001 T1 T2  -> 001 101 111 EOS
+        # 101 T1 PAD -> 101 110 PAD EOS
+        # will have attention scores for the last decoder step:
+        # -inf -inf 1
+        # -inf -inf 1
+        # The mask would set this to
+        # -inf -inf 1
+        # -inf -inf -inf
+        # which results in NaNs
+        attn_containing_nan = attn
+        attn = attn_containing_nan.clone()
+        attn.masked_fill_(attn_containing_nan != attn_containing_nan, 0)
```

Comment:
doesn't `attn_containing_nan != attn_containing_nan` always return False, and then his statement doesn't do anything?